clash-lib/src/Clash/Backend/VHDL.hs

```diff
@@ -131,12 +131,17 @@ instance Backend VHDLState where
   inst            = inst_
   expr            = expr_
   iwWidth         = use intWidth
-  toBV _ id_      = do
-    nm <- Mon $ use modNm
-    pretty (TextS.toLower nm) <> "_types.toSLV" <> parens (pretty id_)
-  fromBV t id_  = do
-    nm <- Mon $ use modNm
-    qualTyName t <> "'" <> parens (pretty (TextS.toLower nm) <> "_types.fromSLV" <> parens (pretty id_))
+  toBV t id_
+    | isBV t = pretty id_
+    | otherwise = do
+      nm <- Mon $ use modNm
+      -- TODO It would be nice to leave out the type qualification if we know id_ is a var
```

Comment:
Is checking whether `id_` is in the `seenIdentifiers` map a way to check if `id_` is a var?

Reply:
`id_ :: Text`
So we can't really do this without changing the API of `toBV`, which I didn't want to do for this PR.

Reply:
`type Identifier = Text`

Reply:
I think @alex-mckenna is right, and we can just check the `id_` against `seenIdentifier`. If the expressions rendered to a string corresponds to a `seenIdentifier` while it shouldn't have then that's a whole different can of worms anyway.